cpp/order/Cubatic.cc

```diff
@@ -155,14 +164,14 @@ Cubatic::Cubatic(float t_initial, float t_final, float scale, unsigned int n_rep
 tensor4 Cubatic::calcCubaticTensor(quat<float>& orientation)
 {
     tensor4 calculated_tensor = tensor4();
-    for (unsigned int i = 0; i < 3; i++)
+    for (auto& m_system_vector : m_system_vectors)
```

Comment:
There is a _lot_ of the code base that could be rewritten using `std::algorithm`. I'll make this particular change as requested, but let's avoid making any more related requests on this PR; we can do that more gradually if and when we have other reasons to modify the relevant parts of the code (as we make other changes cppcheck will probably  automatically start identifying more opportunities as well).

Reply:
Actually changed my mind, I'm skipping this for now. Currently tensor4 only supports in place addition. We could add the extra operator, but the result of making this change would end up being a net increase in lines of code with limited extra clarity anyway since the accumulation involves a nontrivial operation. We can revisit in the future.